### PR TITLE
[mono][interp] Add checks for interpreter stack overflow.

### PR DESCRIPTION
**Pull request by vargaz**
These could show up as random crashes esp. on wasm.

### CODE REVIEW

**Review by radekdoulik — APPROVED** [review]

**Comment by vargaz** [comment]
Added it.

File: src/mono/mono/mini/interp/interp.c [hunk 1]
```diff
@@ -3703,6 +3709,11 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 			 * than the callee stack frame (at the interp level)
 			 */
 			context->stack_pointer = (guchar*)frame->stack + new_method->alloca_size;
+			if (G_UNLIKELY (context->stack_pointer >= context->stack_end)) {
+				context->stack_end = context->stack_real_end;
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/interp.c:3713]: I don't think we should be changing `stack_end`.

Comment (vargaz) [reply src/mono/mono/mini/interp/interp.c:3713]: Without it, the other asserts will be hit during handling the exception.

Comment (BrzVlad) [reply src/mono/mono/mini/interp/interp.c:3713]: Maybe we could change the `stack_end` here, in order to allow some more space for filters and finally to run if they need more stack for whatever reason, but set back the `stack_end` to the original value in order to preserve functionality for multiple stack overflows. (I think we can reset it in `interp_set_resume_state`). In that case, asserting for the real end might make more sense.

Comment (vargaz) [reply src/mono/mono/mini/interp/interp.c:3713]: Going to merge this for now since it might help diagnose
https://github.com/dotnet/runtime/issues/65335

**Pull request merged**